멘토님 코드 리뷰용 PR

kdo1999 opened the pull request:
<!--
  템플릿은 아직 PR 작성이 익숙하지 않으신 분들을 위해서 제공하는 가이드입니다!
  리뷰어 또는 이 PR을 보게 될 다른 사람들이 이 PR을 보는데 참고할 수 있는 내용이 있다면 포함해서 작성해주시면 됩니다.
-->

## 📌 과제 설명 <!-- 어떤 걸 만들었는지 대략적으로 설명해주세요 -->

## 👩‍💻 요구 사항과 구현 내용 <!-- 기능을 Commit 별로 잘개 쪼개고, Commit 별로 설명해주세요 -->

## ✅ 피드백 반영사항  <!-- 지난 코드리뷰에서 고친 사항을 적어주세요. 재PR 시에만 사용해 주세요! (재PR 아닌 경우 삭제) -->

## ✅ PR 포인트 & 궁금한 점 <!-- 리뷰어 분들이 집중적으로 보셨으면 하는 내용을 적어주세요 -->

vdvhk12 committed:
feat: 낚시 포인트에 location 필드 추가 (MIKKI-105)

vdvhk12 committed:
feat: 지도 바운드 내 낚시 포인트 조회 API 구현 (MIKKI-105)

- ST_GeomFromText를 이용한 공간 쿼리 적용
- QueryDSL로 FishPointResponse DTO 조회
- Controller, Service, Repository 레이어 구성 완료

vdvhk12 committed:
feat: 지역명 기준 낚시 포인트 검색 기능 추가 (MIKKI-105)

- fishPointName에 대해 부분 일치 검색 (LIKE)
- isBan = false인 낚시 포인트만 조회
- 추후 DTO + @Valid 방식으로 리팩토링 예정

vdvhk12 committed:
refactor: DTO + @Valid 방식으로 쿼리 파라미터 검증 (MIKKI-105)

vdvhk12 committed:
feat: 내 위치 기반 반경 내 낚시 포인트 조회 기능 구현 (MIKKI-105)

- ST_Distance를 활용한 반경 필터링 적용
- 거리(km)를 함께 응답에 포함
- isBan 조건 추가하여 금지 포인트 제외

vdvhk12 committed:
feat: 내 위치 기준 가장 가까운 낚시 포인트 3개 조회 API 구현 (MIKKI-105)

- ST_Distance를 활용해 거리 계산
- 가까운 순으로 정렬 후 상위 3개 limit 조회

vdvhk12 committed:
feat: region 엔티티 추가 & fishPoint 엔티티에 regionId 컬럼 추가 (MIKKI-105)

kdo1999 committed:
test: 실패 테스트 수정 (MIKKI-136)

kdo1999 committed:
comment: 주석 추가 (MIKKI-136)

kdo1999 committed:
refactor: 선박 저장 메소드명 수정 (MIKKI-136)

Account-4-Review commented:
리뷰 방향성 미작성으로, 임의대로 리뷰합니다. 참고해주세요.

kangjjs committed:
refactor : 누락된 @Enumerated 추가 (MIKKI-137)

kangjjs committed:
refactor : 에러코드 수정 (MIKKI-137)

kangjjs committed:
Merge pull request #49 from prgrms-web-devcourse-final-project/feature/MIKKI-137

MIKKI-137

kdo1999 commented:
@Account-4-Review 커서 적용했는데 이런 방식을 말씀하셨던게 맞는지 또는 개선이 더 필요한지 궁금합니다.

작업 수정사항이 많이 생겼어서 말씀 드린다는게 깜빡했습니다 ㅠㅠ

kdo1999 committed:
refactor: 코드 리뷰 반영 (MIKKI-136)

kdo1999 committed:
Merge pull request #51 from prgrms-web-devcourse-final-project/feature/MIKKI-136

MIKKI-136

Account-4-Review commented on backend/src/main/java/com/backend/domain/fish/repository/FishQueryRepository.java:
```diff
@@ -0,0 +1,92 @@
+	private final JPAQueryFactory jpaQueryFactory;
+	private final JdbcTemplate jdbcTemplate;
+
+	public Optional<FishResponse.Detail> findDetailById(final Long fishId) {
```
가능하면 쿼리에 인덱스를 명시해 주세요.

Account-4-Review commented on backend/src/main/java/com/backend/domain/fish/repository/FishQueryRepository.java:
```diff
@@ -0,0 +1,92 @@
+
+			// JDBC batch update 사용
+			String sql = "UPDATE fishes SET fishes.popularity_score = ? WHERE fishes.fish_id = ?";
+			jdbcTemplate.batchUpdate(sql, hourlyFishCountSummaryList, hourlyFishCountSummaryList.size(),
```
지금 구조로는 데이터가 수만개, 수십만개가 되어도 그냥 무조건 update를 한방에 때릴 것 같은데, 다소 위험해 보이네요.

Account-4-Review commented on backend/src/main/java/com/backend/domain/fishencyclopedia/repository/FishEncyclopediaQueryRepository.java:
```diff
@@ -1,66 +1,229 @@
+	) {
+		try {
+			switch (sortField) {
+				case "count" -> {
```
보통 함수 참조로 빼고, 별도의 함수로 분리하는게 코드 관리 측면에서 더 나을 것 같아요.

Account-4-Review commented on backend/src/main/java/com/backend/domain/fishencyclopedia/service/FishEncyclopediaServiceImpl.java:
```diff
@@ -52,16 +57,21 @@ public Long createFishEncyclopedia(final FishEncyclopediaRequest.Create requestD
+			.orElse(CatchMaxLengthConvert.fromCatchMaxLength(requestDto.fishId(), memberId));
+
+ 		// 최대 값 갱신 & 잡은 마리 수 설정
+		catchMaxLength.setBestLength(requestDto.length());
```
BestLength, CatchCount 는 따로 따로 업데이트 되는 경우가 없죠.
하나로 묶어주세요.

Account-4-Review commented on backend/src/main/java/com/backend/domain/fish/scheduler/FishScheduler.java:
```diff
@@ -0,0 +1,38 @@
+	 *
+	 * @author Kim Dong O
+	 */
+	@Scheduled(cron = "0 0 * * * *", zone = "Asia/Seoul")
```
트래픽이 없는 새벽 시간대에는 어떻게 이 로직을 커버해야 할까요?

Account-4-Review commented on backend/src/main/java/com/backend/domain/fishingtrippost/service/FishingTripPostService.java:
```diff
@@ -0,0 +1,60 @@
+	 * @return 게시글 상세 정보가 담긴 {@link FishingTripPostResponse.Detail}
+	 * @throws FishingTripPostException 게시글이 존재하지 않는 경우 발생
+	 */
+	FishingTripPostResponse.Detail getFishingTripPostDetail(final Long fishingTripPostId);
```
이미지가 없는 경우에 대한 예외는 있나요?

Account-4-Review commented on backend/src/main/java/com/backend/domain/fishingtriprecruitment/repository/FishingTripRecruitmentQueryRepository.java:
```diff
@@ -0,0 +1,71 @@
+		final Long fishingTripPostId,
+		final RecruitmentStatus status
+	) {
+		List<FishingTripRecruitmentResponse.DetailPage> detailPageList = jpaQueryFactory.select(
```
멘토링 시간때 봅시다.

Account-4-Review commented on backend/src/main/java/com/backend/domain/fishingtriprecruitment/service/FishingTripRecruitmentServiceImpl.java:
```diff
@@ -0,0 +1,189 @@
+	private FishingTripPost validateFishingTripPostOwner(final Long memberId, final Long fishingTripPostId) {
+		FishingTripPost fishingTripPost = fishingTripPostRepository.findById(fishingTripPostId)
+			.orElseThrow(() -> new FishingTripPostException(FishingTripPostErrorCode.FISHING_TRIP_POST_NOT_FOUND));
+		log.debug("[동출 모집 게시글 조회] : {}", fishingTripPost);
```
매번 코드에 넣기도 힘들 것 같네요 ㅋㅋㅋ
나중에 이거 제거하는 것도 일이될 것 같은데...
사실 저도 이정도로 매번 다 넣을 줄은 몰랐는데 말이죠.
차라리 AOP 기반으로 작업하는게 맞지 않았을까 생각이 듭니다.

kdo1999 committed:
feat: 물고기 전체 조회 추가 (MIKKI-144)
- Repository

kdo1999 committed:
feat: 물고기 전체 조회 추가 (MIKKI-144)
- Service

kdo1999 committed:
feat: 물고기 전체 조회 추가 (MIKKI-144)
- Controller

kdo1999 committed:
feat: 물고기 전체 조회 응답 객체 추가 (MIKKI-144)

kdo1999 committed:
feat: Swagger 어노테이션 추가 (MIKKI-144)

kdo1999 committed:
comment: 주석 제거 (MIKKI-144)

Account-4-Review commented on backend/src/main/java/com/backend/domain/fishingtriprecruitment/service/FishingTripRecruitmentServiceImpl.java:
```diff
@@ -0,0 +1,189 @@
+
+	@Override
+	@Transactional
+	public void acceptFishingTripRecruitment(final Long memberId, final Long fishingTripRecruitmentId) {
```
잠재적인 동시성 이슈가 있어보이네요.

kdo1999 committed:
feat: EnableScheduling 추가

vdvhk12 committed:
feat: regionId로 낚시 포인트 조회 기능 추가 (MIKKI-105)

swjoon committed:
remove: 어류 & 선상낚시 게시글 중간 테이블 삭제 (MIKKI-138)

vdvhk12 committed:
refactor: 낚시 포인트 엔티티에 columnDefinition = "POINT SRID 4326" 옵션 제거 (MIKKI-105)

- h2 에서 지원하지 않아서 일단 지우고, DB 단에서 직접 설정

vdvhk12 committed:
refactor: 누락된 final 추가 (MIKKI-105)

vdvhk12 committed:
fix: 동출 게시글 테스트 낚시 포인트 생성 메서드 수정 (MIKKI-105)

- 낚시 포인트 생성하는 메서드 util 클래스에 작성

vdvhk12 committed:
refactor: 사용하지 않는 메서드 제거 (MIKKI-105)

swjoon committed:
refactor: 선상낚시 게시글 컬럼명 수정 및 index 설정 (MIKKI-138)

vdvhk12 committed:
refactor: 응답 dto 객체 네이밍 수정 (MIKKI-105)

swjoon committed:
feat: response, request 컬럼명 수정 & 추가 (MIKKI-138)

swjoon committed:
refactor: 선상 낚시 게시글 조회 및 검색 메서드 리펙토링 (MIKKI-138)
- Repository
- cursor 적용

swjoon committed:
refactor: 선상 낚시 게시글 조회 및 검색 메서드 리펙토링 (MIKKI-138)
- Service
- image 적용 dto 조립

swjoon committed:
refactor: 선상 낚시 게시글 조회 및 검색 메서드 리펙토링 (MIKKI-138)
- Controller
- url 수정

swjoon committed:
feat: QuerydslUtil Long, Double 타입 추가 (MIKKI-138)

swjoon committed:
test: 선상 낚시 게시글 조회 및 검색 메서드 테스트 코드 재작성 (MIKKI-138)

swjoon committed:
test: 선상 낚시 게시글 조회 및 검색 메서드 테스트 코드 재작성 (MIKKI-138)

swjoon committed:
feat: 선상 낚시 게시글 에러코드 추가 (MIKKI-138)

vdvhk12 committed:
feat: 인기 낚시 포인트 조회 기능 추가 (MIKKI-105)

- 서비스 메서드에 @Transactional(readOnly = true) 추가

swjoon committed:
feat: 선상 낚시 게시글 생성 검증 추가(MIKKI-138)
- 최대 승선인원 검증

swjoon committed:
test: 선상 낚시 게시글 생성 검증 테스트 코드 작성(MIKKI-138)
- 최대 승선인원 검증 테스트

vdvhk12 committed:
refactor: 낚시 포인트 조회에 notBanned 조건 추가 (MIKKI-105)

swjoon committed:
rename: 패키지 이름 수정(MIKKI-138)

kdo1999 committed:
Merge pull request #52 from prgrms-web-devcourse-final-project/feature/MIKKI-144

MIKKI-144

swjoon committed:
refactor: 리뷰 내용 반영 (MIKKI-138)
- 누락된 final
- from -> converter 이동

kdo1999 committed:
Merge pull request #54 from prgrms-web-devcourse-final-project/feature/MIKKI-138

MIKKI-138

vdvhk12 committed:
refactor: 누락된 fianl 추가 (MIKKI-105)

vdvhk12 committed:
refactor: 이름과 지역id에 복합 인덱스 추가 (MIKKI-105)

kangjjs committed:
Merge pull request #53 from prgrms-web-devcourse-final-project/feature/MIKKI-105

MIKKI-105

kdo1999 committed:
refactor: 물고기 도감 수정

kdo1999 committed:
fix: 오류 수정
- JwtAuthenticationFilter Exception 메세지가 아닌 cause 로그에 찍는 방법으로 변경
- 물고기 도감 컨버터 수정

kangjjs committed:
feat : 동출 모집완료시 메일전송 기능 구현 (MIKKI-145)

kangjjs committed:
feat : mailConfig 작성 (MIKKI-145)

kangjjs committed:
refactor : setQueueCapacity 20으로 조정 (MIKKI-145)

kangjjs committed:
chore : mail 라이브러리 추가 (MIKKI-145)

kangjjs committed:
feat : 모집완료시 신청완료 메일전송 기능 구현 (MIKKI-145)

kangjjs committed:
feat : 비동기를 위해서 @EnableAsync 추가 (MIKKI-145)

kangjjs committed:
feat : 회원 Id 리스트로 이메일 리스트 받아오는 기능 구현 (MIKKI-145)

kangjjs committed:
feat : 동출게시글에서 승인된 신청자들 id 가져오는 기능 구현 (MIKKI-145)

kangjjs committed:
feat : 동출모집 종료되었을때 신청된 사용자에한해서 메일보내는 기능 추가 (MIKKI-145)

kangjjs committed:
feat : 동출게시글 모집종료 기능구현 (MIKKI-145)

kangjjs committed:
feat : 동출게시글 모집종료 기능 구현 (MIKKI-145)

kangjjs committed:
test : 동출게시글 모집종료 기능구현 (MIKKI-145)

vdvhk12 committed:
feat: 지역 정보 조회 기능 추가 (MIKKI-105)

kangjjs committed:
test : 기존테스트에 게시글모집 완료로직 추가 (MIKKI-145)

kangjjs committed:
test : 사용자 id로 email받아오는 테스트 코드 작성 (MIKKI-145)

kdo1999 committed:
Update build-test.yml

vdvhk12 committed:
refactor: 지역 정보 조회 컨트롤러 별도로 분리 (MIKKI-105)

kdo1999 committed:
Update build-test.yml

kangjjs committed:
Merge pull request #55 from prgrms-web-devcourse-final-project/feature/MIKKI-145

MIKKI-145

vdvhk12 committed:
chore: 주석 추가 (MIKKI-105)

vdvhk12 committed:
refactor: 누락된 final 추가 (MIKKI-105)

vdvhk12 committed:
refactor: response dto 필드 수정 (MIKKI-105)

kdo1999 committed:
Merge pull request #56 from prgrms-web-devcourse-final-project/feature/MIKKI-105

MIKKI-105

Account-4-Review commented on backend/src/main/java/com/backend/domain/fishencyclopedia/repository/FishEncyclopediaQueryRepository.java:
```diff
@@ -1,66 +1,231 @@
+				)
+			)
+			.from(fish)
+			.leftJoin(catchMaxLength)
```
이중조인 부터는 쿼리 실행 시, 임시테이블 구조를 가져가요.
(임시테이블: 실제 테이블이 아니라, 쿼리 실행 중간 단계에서 생성된 가상의 테이블. 메모리 위에 있으나, 크기가 커지거나 램이 부족하면 하드에 써집니다. 이때 성능이 심각하게 저하될 수 있어요.)

MySQL의 실행계획을 확인해보고, 이게 커버링 인덱스 or 임시테이블을 최소화 하여 수행할 수 있는 쿼리인지 확인해보고,
그렇지 않다면 과감하게 쿼리 두방으로 나눠주세요.

Account-4-Review commented on backend/src/main/java/com/backend/domain/fishingtrippost/controller/FishingTripPostController.java:
```diff
@@ -0,0 +1,89 @@
+
+		Long saveFishingTripPostId = fishingTripPostService.createFishingTripPost(user.getId(), requestDto);
+
+		return ResponseEntity.created(URI.create(saveFishingTripPostId.toString()))
```
HTTP Specification 에 따르면, CREATED는 **생성된 자원의 위치나 자원의 값 중 하나를 명시** 하도록 되어 있습니다.
body 를 반환했으면 URI를 던지지 않는게 스펙이에요.

Account-4-Review commented on backend/src/main/java/com/backend/global/email/service/EmailServiceImpl.java:
```diff
@@ -0,0 +1,32 @@
+	private final JavaMailSender mailSender;
+
+	@Override
+	@Async("threadPoolTaskExecutor")
```
여러 게시글이 완료가 동시에 되었다고 생각해 볼게요.
각각의 게시글은 40명 정도의 멤버가 있다고 생각해 봅시다.
-> 게시글이 20개일 때, 현재 로직으로는 800개의 동시 처리 Task가 발생합니다.
메일 발송이 엄청 timely 한 작업이 아니기도 해서, 이 데이터가 큐를 채우는 것 보단 **하나의 게시글 단위 작업** 이 하나의 Task로 처리되도록 작업하는게 더 좋을 것 같아요.

vdvhk12 committed:
feat: 낚시 포인트에 대한 물고기 정보 집계 기능 구현 (MIKKI-105)

- 낚시 포인트에서 가장 많이 잡힌 물고기 4개 조회 기능 구현

vdvhk12 committed:
feat: 낚시 포인트 물고기 정보 서비스 메서드 구현 (MIKKI-105)

- 누락된 final 추가

swjoon committed:
feat: 물고기 조회 메서드 생성 (MIKKI-138-1)

swjoon committed:
fix: GlobalRequest 주석 수정 (MIKKI-138-1)

swjoon committed:
fix: parameter 이름, 조건 수정 (MIKKI-138-1)

swjoon committed:
fix: 예약 일자 수정 및 스웨거 테스트 완료 (MIKKI-138-1)

swjoon committed:
fix: 예약 수정 및 스웨거 테스트 완료 (MIKKI-138-1)

swjoon committed:
fix: 선상 낚시 수정 및 스웨거 테스트 완료 (MIKKI-138-1)

swjoon committed:
test: 테스트 코드 수정 (MIKKI-138-1)
- ReservationDate
- Reservation
- ShipFishingPost

vdvhk12 committed:
feat: 어류 도감 낚시 포인트, 물고기 ID로 집계하는 메서드 추가 (MIKKI-105)

vdvhk12 committed:
feat: 낚시 포인트 별 물고기 집계 테이블 갱신 스케줄링 추가 (MIKKI-105)

- 1시간 마다 낚시 도감에 추가된 물고기가 잡힌 총 개수를 낚시 포인트 집계 테이블에 업데이트

vdvhk12 committed:
feat: 낚시 포인트 상세 조회 기능 추가 (MIKKI-105)

kdo1999 committed:
refactor: description 수정 (MIKKI-149)

kdo1999 committed:
feat: 회원 ID로 저장된 선박 전체 조회 기능 구현 (MiKKI-149)

kdo1999 committed:
feat: 응답용 객체 추가 (MIKKI-149)

kdo1999 committed:
feat: 선박 전체 조회 추가 (MIKKI-149)

vdvhk12 committed:
refactor: 서비스 메서드에 트랜잭션 어노테이션 추가 (MIKKI-105)

vdvhk12 committed:
test: 낚시 포인트 집계 메서드 테스트 추가 (MIKKI-105)

kdo1999 committed:
test: 테스트 수정 (MIKKI-149)

kdo1999 committed:
feat: 생성 시간 응답 필드 추가 (MIKKI-138)

kdo1999 committed:
Merge pull request #58 from prgrms-web-devcourse-final-project/feature/MIKKI-149

MIKKI-149

kdo1999 committed:
refactor: example값 수정 (MIKKI-138)

kangjjs committed:
style : 주석 수정 (MIKKI-151)

kangjjs committed:
feat : id로 file 찾는 기능구현 (MIKKI-151)

kangjjs committed:
feat : 게시글에 regionId 필드 추가 (MIKKI-151)

kangjjs committed:
feat : 동출 게시글 전체보기 (MIKKI-151)

kangjjs committed:
feat : pageDetail로 변환 기능구현 (MIKKI-151)

kangjjs committed:
feat : 동출게시글 전체보기 기능구현 (MIKKI-151)

kangjjs committed:
feat : dto로 나누어 응답추가 (MIKKI-151)

kangjjs committed:
test : 동출게시글 전체보기 테스트 코드작성 (MIKKI-151)

kdo1999 committed:
refactor: ShipRequest.Form 객체명 수정 (MIKKI-155)

kdo1999 committed:
Merge branch 'dev' into feature/MIKKI-138-1

kangjjs committed:
refactor : regionid 필드 추가 (MIKKI-151)

kangjjs committed:
feat : poststatus param값 제대로 안오던거 수정 (MIKKI-151)

kangjjs committed:
test : regionid 필드값 추가 (MIKKI-151)

kangjjs committed:
refactor : final 누락 (MIKKI-151)

kdo1999 committed:
Merge pull request #57 from prgrms-web-devcourse-final-project/feature/MIKKI-138-1

MIKKI-138-1

kdo1999 committed:
Merge pull request #59 from prgrms-web-devcourse-final-project/feature/MIKKI-151

MIKKI-151

vdvhk12 committed:
test: 낚시 포인트 리포지토리 테스트 추가 - 공간 쿼리 제외 (MIKKI-105)

vdvhk12 committed:
test: 낚시 포인트 서비스 테스트 추가 - 공간 쿼리 제외 (MIKKI-105)

kdo1999 committed:
feat: 선박 수정 기능 구현 (MIKKI-155)

kdo1999 committed:
feat: ShipErrorCode 작성자가 아닐 때 예외 코드 추가 (MIKKI-155)

kdo1999 committed:
feat: 선박 수정 기능 구현 (MIKKI-155)

kdo1999 committed:
feat: 선박 수정 기능 구현 (MIKKI-155)

vdvhk12 committed:
test: 낚시 포인트 컨트롤러 테스트 추가 - 공간 쿼리 제외 (MIKKI-105)

kdo1999 committed:
refactor: 코드 리뷰 반영

kdo1999 committed:
refactor: equals 수정 (MIKKI-155)

vdvhk12 committed:
feat: 지역 정보 조회에 위도 경도 필드 추가 제외 (MIKKI-105)

kdo1999 committed:
Merge pull request #60 from prgrms-web-devcourse-final-project/feature/MIKKI-155

MIKKI-155

vdvhk12 committed:
refactor: 낚시 포인트 집계 서비스 코드 가독성 향상을 위해 메서드 분리 (MIKKI-105)

vdvhk12 committed:
refactor: 누락된 final 추가 (MIKKI-105)

vdvhk12 committed:
refactor: 낚시 포인트 조회 검증 메서드 분리 (MIKKI-105)

vdvhk12 committed:
Merge pull request #61 from prgrms-web-devcourse-final-project/feature/MIKKI-105

MIKKI-105

kdo1999 committed:
feat: 선박 삭제시 사용할 에러코드 정의 (MIKKI-156)

kdo1999 committed:
feat: 선박 ID 기준으로 게시글이 존재하는지 기능 구현 (MIKKI-156)

kdo1999 committed:
feat: 선박 삭제 기능 구현 (MIKKI-156)
- Repository

kdo1999 committed:
feat: 선박 삭제 기능 구현 (MIKKI-156)
- Service

kdo1999 committed:
feat: 선박 삭제 기능 구현 (MIKKI-156)
- Controller

kdo1999 committed:
feat: Comment 엔티티 추가 (MIKKI-159)

kdo1999 committed:
feat: 댓글 저장 기능 구현 (MIKKI-159)

swjoon committed:
refactor: 잔여 예약 조회 리펙토링 (MIKKI-131)

swjoon committed:
test: 잔여 예약 조회 리펙토링 (MIKKI-131)

Janghyeonsuk committed:
feat: RedisUtil 구현(MIKKI-120)

Janghyeonsuk committed:
feat: Redis 캐싱 처리 추가(MIKKI-120)

Janghyeonsuk committed:
feat: 게시글 엔티티에 like count 필드 추가 (MIKKI-120)

Janghyeonsuk committed:
feat: 동출 모집 좋아요수 업데이트 구현(MIKKI-120)

Janghyeonsuk committed:
feat: 선상 낚시 좋아요수 업데이트 구현(MIKKI-120)

Janghyeonsuk committed:
feat: 좋아요 엔티티 구현(MIKKI-120)

Janghyeonsuk committed:
feat: 좋아요 dto, domain 구현(MIKKI-120)

Janghyeonsuk committed:
feat: 좋아요 converter 구현(MIKKI-120)

Janghyeonsuk committed:
feat: 좋아요 repository 구현(MIKKI-120)

Janghyeonsuk committed:
feat: 좋아요 service, cache service 구현(MIKKI-120)

Janghyeonsuk committed:
feat: 좋아요 controller 구현(MIKKI-120)

Janghyeonsuk committed:
feat: 좋아요 캐싱값 DB 반영, 소프트 딜리트 clean up 스케줄러 구현(MIKKI-120)

kdo1999 committed:
feat: 댓글 존재 여부 체크, 자식 카운트 업데이트 기능 추가 (MIKKI-159)

kdo1999 committed:
feat: 댓글 생성용 Request 객체 추가 (MIKKI-159)

kdo1999 committed:
feat: 댓글 에러 코드 추가 (MIKKI-159)

kdo1999 committed:
feat: 댓글 저장 기능 구현 (MIKKI-159)
- 부모 댓글 검증 및 업데이트 로직 추가

kdo1999 committed:
feat: CommentConverter 추가 (MIKKI-159)

kdo1999 committed:
feat: 댓글 생성 기능 추가 (MIKKI-159)
- Controller

kdo1999 committed:
refactor: 코드 리뷰 반영 (MIKKI-156)

kdo1999 committed:
Merge pull request #62 from prgrms-web-devcourse-final-project/feature/MIKKI-156

MIKKI-156

kdo1999 committed:
Update build-test.yml

swjoon committed:
feat: Scheduler 작업 내역 도메인 생성 (MIKKI-131)
- 작업 기록 가능

swjoon committed:
feat: 리뷰 삭제시 평점 즉시 반영 적용 (MIKKI-131)

swjoon committed:
feat: 선상 낚시 게시글 기능 추가 (MIKKI-131)
- 마이페이지
- 게시글 업데이트
- 평점 스케쥴러

swjoon committed:
test: 리뷰 삭제시 평점 즉시 반영 적용 (MIKKI-131)
- 테스트 코드 적용

kdo1999 committed:
refactor: final 누락 수정 (MIKKI-159)

swjoon committed:
feat: 예약 도메인 기능 추가 및 리펙토링 (MIKKI-131)
- 선장, 유저 예약 목록조회 기능 리펙토링 및 추가
- 지난예약 수정 및 취소 불가 메서드 추가

swjoon committed:
feat: 예약 일자 도메인 기능 추가 및 리펙토링 (MIKKI-131)
- 예약 수정 기능 추가
- 지난예약 검색, 수정 및 취소 불가 메서드 추가

kdo1999 committed:
refactor: 코드리뷰 반영 (MIKKI-159)

kangjjs committed:
feat : 조회 부분 permitAll (MIKKI-126)

kangjjs committed:
feat : 게시글 참여자 & dto 기능 구현 (MIKKI-126)

kangjjs committed:
feat : 동출 게시글 인원조회 기능구현 (MIKKI-126)

kangjjs committed:
feat : 인원조회 response 생성 (MIKKI-126)

kangjjs committed:
test : 동출게시글 인원상세 테스트코드 작성 (MIKKI-126)

kangjjs committed:
refactor : 날짜 set으로 추가 (MIKKI-126)

kdo1999 committed:
refactor: 코드 리뷰 반영 (MIKKI-120)

kdo1999 committed:
refactor: 코드 리뷰 반영 (MIKKI-120)

kdo1999 committed:
Merge branch 'dev' into feature/MIKKI-120

kdo1999 committed:
Merge pull request #63 from prgrms-web-devcourse-final-project/feature/MIKKI-120

MIKKI-120

kangjjs committed:
Merge branch 'dev' into feature/MIKKI-126

kangjjs committed:
Merge pull request #65 from prgrms-web-devcourse-final-project/feature/MIKKI-126

MIKKI-126

kangjjs committed:
Update FishingTripPostRepository.java

vdvhk12 committed:
feat: 마이페이지 내가 수집한 어종 개수 조회 기능 구현 (MIKKI-160)

kangjjs committed:
Merge pull request #64 from prgrms-web-devcourse-final-project/feature/MIKKI-159

MIKKI-159

vdvhk12 committed:
test: 내가 등록한 어종 개수 조회 기능 테스트 추가 (MIKKI-160)

swjoon committed:
test: 예약 일자, 예약, 게시글 도메인 테스트코드 작성 (MIKKI-131)

swjoon committed:
fix: 병합 충돌 해결 (MIKKI-131)

swjoon committed:
feat: 예약 횟수 조회 메서드 생성 (MIKKI-131)

swjoon committed:
feat: 예약 횟수 조회 메서드 테스트 코드 작성 (MIKKI-131)

kangjjs committed:
test : 좋아요 추가한 동출게시글 상세보기 테스트코드 작성 (MIKKI-163)

kangjjs committed:
feat : response에 좋아요 추가 (MIKKI-163)

kangjjs committed:
feat : 인증되지 않은 회원도 조회가능하도록 처리 (MIKKI-163)

kangjjs committed:
feat : 인덱스 추가 (MIKKI-163)

kangjjs committed:
feat : 동출게시글 상세보기 좋아요 추가 (MIKKI-163)

swjoon committed:
feat: 선장 대시보드 조회 메서드 (MIKKI-131)

swjoon committed:
test: 선장 대시보드 조회 메서드 테스트 코드 작성(MIKKI-131)

kangjjs committed:
Merge pull request #66 from prgrms-web-devcourse-final-project/feature/MIKKI-160

MIKKI-160

kdo1999 committed:
feat: 댓글 조회 기능 구현 (MIKKI-165)

vdvhk12 committed:
feat: 채팅방 엔티티 추가 (MIKKI-166)

vdvhk12 committed:
feat: 채팅방 생성 기능 추가 (MIKKI-166)

kangjjs committed:
fix : 테스트 에러 수정 (MIKKI-163)

kdo1999 committed:
test: 테스트 실패 수정 (MIKKI-165)

kdo1999 committed:
comment: 주석 추가 (MIKKI-165)

kangjjs committed:
fix : build 에러 나는거 수정 (MIKKI-163)

kangjjs committed:
fix : buildtest 에러 해결 (MIKKI-163)

vdvhk12 committed:
test: 채팅방 생성 기능 테스트 추가 (MIKKI-166)

kangjjs committed:
Merge pull request #67 from prgrms-web-devcourse-final-project/feature/MIKKI-163

MIKKI-163

kdo1999 committed:
feat: 댓글 조회 기능 추가 (MIKKI-165)

swjoon committed:
Merge pull request #68 from prgrms-web-devcourse-final-project/feature/MIKKI-131

MIKKI-131

kdo1999 committed:
feat: 댓글 조회 기능 추가 (MIKKI-165)

kdo1999 committed:
feat: Schema 추가 (MIKKI-165)

vdvhk12 committed:
feat: 동출 게시글 생성 시 채팅방 생성 로직 추가 (MIKKI-166)

kdo1999 committed:
refactor: 코드 리뷰 반영 (MIKKI-165)

kdo1999 committed:
test: 테스트 실패 수정 (MIKKI-165)

kdo1999 committed:
refactor: 코드 리뷰 반영 (MIKKI-165)

kdo1999 committed:
test: displayname 추가 (MIKKI-165)

kdo1999 committed:
test: 테스트 실패 수정 (MIKKI-165)

kdo1999 committed:
test: 테스트 실패 수정 (MIKKI-165)

kdo1999 committed:
feat: Index 설정 추가 (MIKKI-165)

kdo1999 committed:
Merge pull request #69 from prgrms-web-devcourse-final-project/feature/MIKKI-165

MIKKI-165

kdo1999 committed:
Merge pull request #70 from prgrms-web-devcourse-final-project/feature/MIKKI-166

MIKKI-166

kangjjs committed:
test : 동출 게시글 삭제 테스트코드 작성 (MIKKI-167)

kangjjs committed:
feat : 동출게시글 id로 동출신청 삭제기능 구현 (MIKKI-167)

kangjjs committed:
feat : 동출게시글 삭제 기능구현 (MIKKI-167)

kangjjs committed:
Merge branch 'dev' into feature/MIKKI-167

kdo1999 committed:
feat: 누락된 부분 추가 (MIKKI-165)

kdo1999 committed:
Merge pull request #72 from prgrms-web-devcourse-final-project/feature/MIKKI-165

MIKKI-165

kangjjs committed:
Merge pull request #71 from prgrms-web-devcourse-final-project/feature/MIKKI-167

MIKKI-167

kdo1999 committed:
refactor: URI 수정

kdo1999 committed:
test: 실패 테스트 수정 (MIKKI-174)

kdo1999 committed:
feat: content setter 추가 (MIKKI-174)

kdo1999 committed:
test: 댓글 조회 성공 테스트 추가 (MIKKI-174)

kdo1999 committed:
feat: 댓글 수정시 사용할 에러 코드 정의 (MIKKI-174)

kdo1999 committed:
test: 댓글 수정 엔티티 성공 테스트 추가 (MIKKI-174)

kdo1999 committed:
feat: 댓글 수정시 사용할 요청 객체 추가 (MIKKI-174)

kdo1999 committed:
feat: 댓글 수정 기능 구현 (MIKKI-174)

kdo1999 committed:
feat: 댓글 수정 기능 구현 (MIKKI-174)

kdo1999 committed:
test: 댓글 수정 성공 테스트 추가 (MIKKI-174)

kdo1999 committed:
feat: 댓글 수정 기능 구현 (MIKKI-174)

kdo1999 committed:
test: 댓글 수정 성공 테스트 추가 (MIKKI-174)

kdo1999 committed:
refactor: final 누락 수정 (MIKKI-174)

kdo1999 committed:
test: 실패 테스트 수정 (MIKKI-174)

kdo1999 committed:
refactor: 코드 리뷰 반영 (MIKKI-174)

kdo1999 committed:
Update build-test.yml

kdo1999 committed:
Merge pull request #73 from prgrms-web-devcourse-final-project/feature/MIKKI-174

MIKKI-174

kdo1999 committed:
feat: 댓글 삭제 기능 구현 (MIKKI-179)

kdo1999 committed:
test: 댓글 삭제 기능 구현

kdo1999 committed:
feat: 댓글 자식 카운트 증감 기능 구현 (MIKKI-179)

kdo1999 committed:
feat: 댓글 자식 카운트 증감 기능 구현 (MIKKI-179)

kdo1999 committed:
feat: 댓글 삭제 기능 구현 (MIKKI-179)

kdo1999 committed:
feat: 댓글 삭제 기능 구현 (MIKKI-179)

kdo1999 committed:
feat: 동출 게시글 삭제시 댓글도 함께 삭제 (MIKKI-179)

kdo1999 committed:
test: 실패 테스트 수정 (MIKKI-179)

kdo1999 committed:
Merge pull request #74 from prgrms-web-devcourse-final-project/feature/MIKKI-179

MIKKI-179
